src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -768,13 +968,65 @@ public void ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseN
             }
         }
 
-        private void InvokeOnConnectionActivities(ConnectionInfo connectionInfo)
+        /// <summary>
+        /// Invokes the initial on-connect activities if the provided ConnectParams represents the default
+        /// connection.
+        /// </summary>
+        private void InvokeOnConnectionActivities(ConnectionInfo connectionInfo, ConnectParams connectParams)
         {
+            if (!String.Equals(connectParams.Type, ConnectionType.Default))
+            {
+                return;
+            }
+
             foreach (var activity in this.onConnectionActivities)
             {
                 // not awaiting here to allow handlers to run in the background
                 activity(connectionInfo);
             }
         }
+
+        /// <summary>
+        /// Invokes the fianl on-disconnect activities if the provided DisconnectParams represents the default
+        /// connection or is null - representing that all connections are being disconnected.
+        /// </summary>
+        private void InvokeOnDisconnectionActivities(ConnectionInfo connectionInfo)
+        {
+            foreach (var activity in this.onDisconnectActivities)
```

Comment:
Another thing we should backlog: moving these activities to events. This works fine, but it'll be cleaner to do this via events.